cpp/dolfinx/generation/RectangleMesh.cpp

```diff
@@ -197,9 +197,7 @@ mesh::Mesh build_tri(MPI_Comm comm,
 
           switch (diagonal)
           {
-          case DiagonalType::right_left:
-            local_diagonal = DiagonalType::right;
-            break;
+          case DiagonalType::right_left: 
```

Comment:
This bit of code is confusing.

Reply:
We could use an if (x==a or x==b) instead of a switch ?

Reply:
I think that would be better in this case. The present looks like a bug to me, even if it isn't.